src/main/java/pl/tomaszdziurko/jvm_bloggers/mailing/BlogPostForMailItem.java

```diff
@@ -1,29 +1,110 @@
 package pl.tomaszdziurko.jvm_bloggers.mailing;
 
-import lombok.Data;
+import com.google.common.base.Preconditions;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import lombok.Getter;
+import org.springframework.web.util.UriComponentsBuilder;
+import pl.tomaszdziurko.jvm_bloggers.UTMParameters;
 import pl.tomaszdziurko.jvm_bloggers.blog_posts.domain.BlogPost;
 import pl.tomaszdziurko.jvm_bloggers.blogs.domain.Blog;
 
-
-@Data
+/**
+ * {@link BlogPost} representation for email newsletter template purpose.
+ */
+@Getter
 class BlogPostForMailItem {
-
+    
+    private static final String UTM_SOURCE = "jvmbloggers";
+    private static final String UTM_MEDIUM = "newsletter";
+    
     private String title;
     private String url;
     private String authorLabel;
 
-    public BlogPostForMailItem(BlogPost blogPost) {
-        this.title = blogPost.getTitle();
-        this.url = blogPost.getUrl();
-        this.authorLabel = determineAuthorLabel(blogPost.getBlog());
+    /**
+     * Returns internal builder.
+     * 
+     * @return builder
+     */
+    public static Builder builder() {
+        return new Builder();
     }
+    
+    /**
+     * Builder for {@link BlogPostForMailItem}
+     */
+    public static class Builder {
 
-    private String determineAuthorLabel(Blog blogger) {
-        if (blogger.getTwitter() != null) {
-            return "<a href=\"https://twitter.com/" + blogger.getTwitter().substring(1) + "\">" + blogger.getAuthor() + "</a>";
-        } else {
-            return blogger.getAuthor();
+        private final BlogPostForMailItem instance = new BlogPostForMailItem();
+
+        public Builder from(BlogPost blogPost) {
+            instance.title = blogPost.getTitle();
+            instance.url = blogPost.getUrl();
+            instance.authorLabel = determineAuthorLabel(blogPost.getBlog());
+            return this;
+        }
+        
+        public Builder withTitle(String title) {
```

Comment:
withTitle, withUrl and withAuthorLabel are not used and generally their logic is in from() method so we could get rid of them.